lib/rules/function-calc-no-unspaced-operator/index.cjs

```diff
@@ -381,14 +276,74 @@ const rule = (primary, _secondaryOptions, context) => {
 					inMathFunction: false,
 				},
 			);
-
-			if (needsFix) {
-				setDeclarationValue(decl, cssParserAlgorithms.stringify([nodes]));
-			}
 		});
 	};
 };
 
+/**
+ * @param {string} value
+ * @returns {Array<ComponentValue>}
+ */
+function tokenizeDeclarationValue(value) {
+	const tokens = cssTokenizer.tokenize({ css: value });
+
+	// Step 1
+	// Step 1.1
+	// Re-tokenize dimensions with units containing dashes.
+	// These might be typo's.
+	// For example: `10px-20px` has a unit of `px-20px`
+	tokens.forEach((token, i) => {
+		if (!cssTokenizer.isTokenDimension(token)) return;
+
+		const { unit } = token[4];
+
+		if (unit.startsWith('--')) return;
+
+		const indexOfDash = unit.indexOf('-');
+
+		if (indexOfDash === -1) return;
+
+		const remainder = unit.slice(indexOfDash);
+
+		if (remainder.length === 1) return;
+
+		cssTokenizer.mutateUnit(token, unit.slice(0, indexOfDash));
+		token[3] = token[2] + token[1].length;
+
+		const remainderTokens = cssTokenizer.tokenize({ css: remainder }).slice(0, -1); // Trim EOF token
+
+		remainderTokens.forEach((remainderToken) => {
+			remainderToken[2] += token[3];
+			remainderToken[3] += token[3];
+		});
+
+		tokens.splice(i + 1, 0, ...remainderTokens);
+	});
+
+	// Step 1.2
+	// Re-tokenize scss interpolation blocks
+	// Grouping `#` and `{` into a single token allows us to parse these as simple blocks with curly braces.
+	// For example: `#{$foo}`
+	tokens.forEach((currentToken, i) => {
+		if (!cssTokenizer.isTokenDelim(currentToken) || currentToken[4].value !== '#') return;
+
+		const nextToken = tokens[i + 1];
+
+		if (!cssTokenizer.isTokenOpenCurly(nextToken)) return;
+
+		const nextNextToken = tokens[i + 2];
+
+		if (!cssTokenizer.isTokenDelim(nextNextToken) || nextNextToken[4].value !== '$') return;
+
+		// Set the string representation of the open curly to `#{`
+		nextToken[1] = '#{';
+		// Remove the `#` token
+		tokens.splice(i, 1);
+	});
+
+	return cssParserAlgorithms.parseListOfComponentValues(tokens);
+}
+
 /** @see https://drafts.csswg.org/css-values/#typedef-calc-value */
```

Comment:
https://www.w3.org/TR/css-values-4/#typedef-calc-value

Reply:
The draft versions always contain the latest information and are also the documents used by browser implementers.

The published versions are more of a snapshot for historical documentation purposes.
So that you can link back to a previous iteration of the spec.